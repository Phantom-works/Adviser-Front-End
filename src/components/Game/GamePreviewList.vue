<template>
  <div class="game-list">
    <div v-for="game in games" :key="game.id">
      <GamePreview :game="game" />
    </div>
  </div>
</template>

<script lang="ts" setup>
import Game from '@/Model/Game'
import GamePreview from "@/components/Game/GamePreview.vue";

defineProps<{ games: Game[] }>()
</script>

<style scoped>
.game-list {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center; /* center horizontally */
  margin-top: 5%;
}
</style>

