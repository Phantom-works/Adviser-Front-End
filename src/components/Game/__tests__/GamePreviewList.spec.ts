import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import GamePreviewList from '../GamePreviewList.vue'
import type Game from '../../../Model/Game'

describe('GamePreviewList Snapshots', () => {
  // Arrange

  it('Snapshot Empty List', () => {
    //Arrange
    const games = <Game[]>[]
    // Act
    const wrapper = mount(GamePreviewList, {
      props: {
        games: games
      }
    })
    // Assert
    expect(wrapper.html()).toMatchSnapshot()
  })
})
