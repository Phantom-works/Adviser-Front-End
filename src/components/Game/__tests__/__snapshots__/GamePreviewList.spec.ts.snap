// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GamePreviewList Snapshots > Snapshot 3 Games Factorio 1`] = `
"<div data-v-cc4cce50=\\"\\" class=\\"game-list\\">
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"3A45ST6JYDK6U4TAW4RES5T6JDYK7E4YRAUS5TI6DR4YEU5SRTID6Y\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"3A45ST6JYDK6U4TAW4RES5T6JDYK7E4YRAUS5TI6DR4YEU5SRTID6Y\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"3A45ST6JYDK6U4TAW4RES5T6JDYK7E4YRAUS5TI6DR4YEU5SRTID6Y\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
</div>"
`;

exports[`GamePreviewList Snapshots > Snapshot 3 Games empty 1`] = `
"<div data-v-cc4cce50=\\"\\" class=\\"game-list\\">
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
</div>"
`;

exports[`GamePreviewList Snapshots > Snapshot 12 Games empty 1`] = `
"<div data-v-cc4cce50=\\"\\" class=\\"game-list\\">
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
</div>"
`;

exports[`GamePreviewList Snapshots > Snapshot 12 Games factorio 1`] = `
"<div data-v-cc4cce50=\\"\\" class=\\"game-list\\">
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"3A45ST6JYDK6U4TAW4RES5T6JDYK7E4YRAUS5TI6DR4YEU5SRTID6Y\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"3A45ST6JYDK6U4TAW4RES5T6JDYK7E4YRAUS5TI6DR4YEU5SRTID6Y\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"3A45ST6JYDK6U4TAW4RES5T6JDYK7E4YRAUS5TI6DR4YEU5SRTID6Y\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"3A45ST6JYDK6U4TAW4RES5T6JDYK7E4YRAUS5TI6DR4YEU5SRTID6Y\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"3A45ST6JYDK6U4TAW4RES5T6JDYK7E4YRAUS5TI6DR4YEU5SRTID6Y\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"3A45ST6JYDK6U4TAW4RES5T6JDYK7E4YRAUS5TI6DR4YEU5SRTID6Y\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"3A45ST6JYDK6U4TAW4RES5T6JDYK7E4YRAUS5TI6DR4YEU5SRTID6Y\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"3A45ST6JYDK6U4TAW4RES5T6JDYK7E4YRAUS5TI6DR4YEU5SRTID6Y\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"3A45ST6JYDK6U4TAW4RES5T6JDYK7E4YRAUS5TI6DR4YEU5SRTID6Y\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"3A45ST6JYDK6U4TAW4RES5T6JDYK7E4YRAUS5TI6DR4YEU5SRTID6Y\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"3A45ST6JYDK6U4TAW4RES5T6JDYK7E4YRAUS5TI6DR4YEU5SRTID6Y\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
  <div data-v-cc4cce50=\\"\\">
    <div data-v-023eec1f=\\"\\" data-v-cc4cce50=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"3A45ST6JYDK6U4TAW4RES5T6JDYK7E4YRAUS5TI6DR4YEU5SRTID6Y\\" src=\\"\\" class=\\"game-image\\"></div>
  </div>
</div>"
`;

exports[`GamePreviewList Snapshots > Snapshot Empty List 1`] = `"<div data-v-cc4cce50=\\"\\" class=\\"game-list\\"></div>"`;
